### PR TITLE
perf: opt playable lifetime func

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
  - Introduced `ActivationTrack` and `SpriteColorTrack` classes for enhanced timeline and sprite color manipulation.
  - Added new methods and properties for improved animation and frame processing.

- **Bug Fixes**
  - Updated various methods to ensure proper handling of `binding` properties.

- **Refactor**
  - Renamed and reorganized multiple classes and methods for better clarity and maintainability.
  - Changed import paths and export statements to reflect new module structures.

- **Chores**
  - Updated `package.json` dependencies to the latest versions.
  - Commented out unused `butterfly` object properties in `scene-list.ts`.

- **Documentation**
  - Added new export statements for `GizmoComponent` in the editor-gizmo plugin package.

### CODE REVIEW

File: packages/effects-core/src/plugins/particle/particle-vfx-item.ts [hunk 1]
```diff
@@ -9,17 +10,19 @@ import { ParticleSystem } from './particle-system';
 export class ParticleBehaviourPlayable extends Playable {
   particleSystem: ParticleSystem;
 
-  override onPlayablePlay (): void {
-    this.particleSystem = this.bindingItem.getComponent(ParticleSystem)!;
+  override onPlayablePlay (context: FrameContext): void {
+    const binding = context.output.getUserData() as VFXItem;
+
+    this.particleSystem = binding.getComponent(ParticleSystem)!;
```
[the remark this answers is not in the document]
Comment (yiiqii) [reply packages/effects-core/src/plugins/particle/particle-vfx-item.ts:16]: 下面有对 this.particleSystem 做判断，这块就不用用 ！强转换了

[the remark this answers is not in the document]
Comment (yiiqii) [reply packages/effects-core/src/plugins/particle/particle-vfx-item.ts:16]: 另外，！强转是比较危险的类型操作，能不用尽量不用 @wumaolinmaoan

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/plugins/timeline/tracks/transform-track.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { effectsClass } from 'packages/effects-core/src/decorators';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/timeline/tracks/transform-track.ts:1]: 路径问题

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/plugins/timeline/track.ts [hunk 3]
```diff
@@ -1,10 +1,10 @@
+import { ParticleSystem } from '@galacean/effects-core';
```
Comment (yiiqii) [inline packages/effects-core/src/plugins/timeline/track.ts:1]: 路径问题

**Review by yiiqii — CHANGES_REQUESTED** [review]

File: packages/effects-core/src/plugins/particle/particle-vfx-item.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/particle/particle-vfx-item.ts]: 已修改

File: packages/effects-core/src/plugins/timeline/tracks/transform-track.ts (hunk 2 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/timeline/tracks/transform-track.ts]: 已修改

File: packages/effects-core/src/plugins/timeline/track.ts (hunk 3 above)
Comment (wumaolinmaoan) [reply packages/effects-core/src/plugins/timeline/track.ts]: 已修改

**Review by yiiqii — APPROVED** [review]

**Review by yiiqii — APPROVED** [review]

**Pull request merged**